homeassistant/components/yolink/entity.py

```diff
@@ -18,55 +20,67 @@ def resolve_state_in_event(data: BRDP, event_type: str) -> dict | None:
     return None
 
 
-class YoLinkHADevice(YoLinkDevice):
-    """YoLink Device Common."""
+class YoLinkEntity(CoordinatorEntity[YoLinkCoordinator]):
+    """YoLink Device Basic Entity."""
 
     def __init__(
         self,
         hass: HomeAssistant,
-        device: dict,
+        coordinator: YoLinkCoordinator,
+        device_info: dict,
         client: YoLinkClient,
     ) -> None:
-        """HA YoLink Device."""
-        super().__init__(device, client)
+        """Init YoLink Entity."""
+        super().__init__(coordinator)
         self.hass = hass
+        self.device = YoLinkDevice(device_info, client)
 
     @property
     def device_id(self) -> str:
         """Return the device id of the YoLink device."""
-        return self._device_id
+        return self.device.device_id
 
+    async def async_added_to_hass(self) -> None:
+        """Add to hass."""
 
-class YoLinkEntity(YoLinkHADevice, Entity):
-    """YoLink Device Basic Entity."""
+        async def request_state():
+            resp = await self.device.fetch_state_with_api()
+            if "state" in resp.data:
+                self.update_entity_state(resp.data[ATTR_DEVICE_STATE])
+
+        self.hass.create_task(request_state())
+        self.coordinator.async_add_listener(self._handle_coordinator_update)
```

Comment:
We don't need to add this listener if we inherit from CoordinatorEntity. It is done by default.